packages/web3-core-provider/test/unit/initWeb3Provider.test.ts

```diff
@@ -48,15 +48,15 @@ describe('Instantiates correct provider for varying provided clients', () => {
         expect(web3ProvidersHttp.supportsSubscriptions).not.toBe(undefined);
     });
 
-    it('should instantiate WebSocket provider', () => {
+    it('should instantiate WebSocket provider', (done) => {
```

Comment:
I'm assuming `done` was required here because [connect](https://github.com/ChainSafe/web3.js/blob/junaid/wstests/packages/web3-providers-ws/src/index.ts#L417) is called when the constructor calls `setWeb3Client` which is actually trying to connect to the `wsClient`. Instead of doing this and having an attempted connection, make use of the Jest mock functions as used [here](https://github.com/ChainSafe/web3.js/blob/4.x/packages/web3-providers-http/test/unit/setWeb3Client.test.ts#L18)

So you'd having something like:

```typescript
beforeAll(() => {
    Web3ProvidersWs.prototype.connect = jest.fn();
}
```

Reply:
In general we during unit tests should not initiate any socket, any connection at all and always mock such behaviour.

Reply:
you are right but only in case: 
`Mock functions are also known as "spies", because they let you spy on the behavior of a function that is called indirectly by some other code, rather than only testing the output`  and we are not testing output of connect here.

This test is not for connect ( looking successful connection ) but init of few other items.